python/lsst/meas/algorithms/normalizedCalibrationFlux.py

```diff
@@ -31,6 +31,35 @@
 from .sourceSelector import sourceSelectorRegistry
 
 
+class NormalizedCalibrationFluxError(lsst.pipe.base.AlgorithmError):
+    """Raised if Aperture Correction fails in a non-recoverable way.
+
+    Parameters
+    ----------
+    initNSources : `int`
+        Number of sources selected by the fallback source selector.
+    nCalibFluxFlag : `int`
+        Number of selected sources with raw calibration flux flag unset.
+    nRefFluxFlag : `int`
+        Number of selected sources with reference flux flag unset.
+    """
+    def __init__(self, *, initNSources, nCalibFluxFlag, nRefFluxFlag):
+        msg = "There are no valid stars to compute normalized calibration fluxes."
+        msg += (f" Of {initNSources} initially selected sources, {nCalibFluxFlag} have good raw calibration "
+                f"fluxes and {nRefFluxFlag} have good reference fluxes.")
```

Comment:
My only concern is that this doesn't seem like it contains quite the information one needs to reconstruct the failure conditions. `n_sel==0` is the failure condition, but I don't quite know how the two "good" numbers you give in the error relate to that. Maybe that's ok, I'm just not sure.

Reply:
The idea is to tell you which of the selection cuts is causing there to be no sources. My logic is that this is the first thing you would want to know if you encountered this error.